### PR TITLE
[vcpkg baseline][libcaer] Fix and update

### PR DESCRIPTION
**Pull request by dg0yt**
Updates to the latest release.
Makes opencv an optional feature as in https://gitlab.com/inivation/dv/libcaer/-/blob/3.3.15/CMakeLists.txt#L20
Updates license information.
Marks arm windows as unsupported:
- Fixes baseline regression for libcaer:arm64-windows which was hidden behind libusb until #30714:
  ~~~
  D:\buildtrees\libcaer\src\3bab7bba4e-a26167654c.clean\thirdparty\simple-stdatomic\stdatomic.c(36): warning C4013: '_mm_pause' undefined; assuming extern returning int
  ...
  stdatomic.c.obj : error LNK2019: unresolved external symbol _mm_pause referenced in function zenny_atomic_fetch_add8
  ~~~
  This is in a vendored copy of third-party stdatomic, used only for MSVC (https://github.com/zenny-chen/simple-stdatomic-for-VS-Clang?).
- Upstream says that MSVC is not supported out-of-the-box, and there is no mention of arm at all for Windows: https://gitlab.com/inivation/dv/libcaer/-/blob/master/README.Windows#L7-8

--

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: versions/l-/libcaer.json [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "f90f31425f85b3bb2544a34fc0b9247491ecbf2b",
+      "version": "3.3.15",
+      "port-version": 0
+    },
+    {
+      "git-tree": "0fbfdb1371487f33d8fb4d13edb3bef6e4bfe792",
+      "version-date": "2022-07-25",
+      "port-version": 5
+    },
```
Comment (vicroms) [inline versions/l-/libcaer.json:12]:
```suggestion
```
Remove extraneous version

**Review by vicroms — CHANGES_REQUESTED** [review]

**Comment by MonicaLiu0311** [comment]
Feature `opencv` is tested successfully in the following triplet:
x86-windows
x64-windows
x64-windows-static

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by dg0yt** [comment]
@vicroms Please let us get rid of the "Changes requested" quickly. This baseline regression is a major blocker, even for other pending baseline fixes :cry:

**Review by vicroms — APPROVED** [review]

**Comment by dg0yt** [comment]
@vicroms Thanks! I assume you also re-ran my other PR.

**Pull request merged**